aten/src/ATen/native/TensorShape.cpp

```diff
@@ -2141,36 +2141,56 @@ std::vector<Tensor> meshgrid(TensorList tensors,
   int64_t size = tensors.size();
   TORCH_CHECK(size > 0, "meshgrid expects a non-empty TensorList");
 
+  std::vector<std::reference_wrapper<const Tensor>> tensor_refs(tensors.begin(),
+                                                                tensors.end());
+  bool swap_first_and_second_tensors;
+
   if (!indexing.has_value()) {
     TORCH_WARN_ONCE("In an upcoming release, it will be required to pass the "
                     "indexing argument to torch.meshgrid.");
     indexing = "ij";
+    swap_first_and_second_tensors = false;
   } else {
-    TORCH_CHECK(*indexing == "ij", "Unsupported indexing: ", *indexing);
+    if (*indexing == "xy") {
+      // We can only swap if there are multiple tensors.
+      swap_first_and_second_tensors = size >= 2;
+      if (swap_first_and_second_tensors) {
+        std::swap(tensor_refs[0], tensor_refs[1]);
+      }
+    } else {
+      TORCH_CHECK(*indexing == "ij", "Unsupported indexing: ", *indexing);
+      swap_first_and_second_tensors = false;
+    }
   }
 
   std::vector<int64_t> shape(size);
   for(const auto i: c10::irange(size)){
-    switch (tensors[i].dim()) {
+    switch (tensor_refs[i].get().dim()) {
     case 0:
       shape[i] = 1;
       break;
     case 1:
-      shape[i] = tensors[i].size(0);
+      shape[i] = tensor_refs[i].get().size(0);
       break;
     default:
-      AT_ERROR("Expected scalar or 1D tensor in the tensor list but got: ", tensors[i]);
+      AT_ERROR("Expected scalar or 1D tensor in the tensor list but got: ",
+               tensor_refs[i]);
     }
   }
   for(const auto i: c10::irange(size - 1)){
```

Comment:
These checks are great but can they be moved to the top of the function with a comment like "validates all tensors have the same dtype and device"?

Reply:
sure. these were actually already there, but i unnecessarily switched them to operate on the tensor_refs vector. switched back to the input vector and moved closer to the top. much better.

Reply:
I understand, but while we're here we might as well spruce up the code